packages/fluent-theme/src/components/sendBox/private/useSubmitError.ts

```diff
@@ -37,9 +38,21 @@ const useSubmitError = ({
     [localize]
   );
 
-  return useMemo<Readonly<[RefObject<SendError | undefined>, string | undefined]>>(
-    () => Object.freeze([submitErrorRef, submitErrorRef.current && errorMessageStringMap.get(submitErrorRef.current)]),
-    [errorMessageStringMap, submitErrorRef]
+  const hasValue = !!message?.trim();
+
+  useEffect(() => {
+    if (error === 'empty' && hasValue) {
+      setError(undefined);
+    }
+  }, [error, hasValue]);
+
+  const checkError = useCallback(() => {
+    setError(submitErrorRef.current);
+  }, [submitErrorRef]);
+
+  return useMemo<Readonly<[RefObject<SendError | undefined>, string | undefined, () => void]>>(
```

Comment:
The return value of this hook feels like a random choice of stuff.

Reply:
It only looks like. The ref contains the actual error state, while the string contains the error state we'd like to show. The ref is required to be ref so we don't accidentally re-render on the change.

Sorry, I'm not seeing better design at this point. I probably can rename things to make this clear.

Reply:
Updated. It's not perfect still, but should look a bit better.